File: framework/csp1d-interface-renderer/src/components/dto.ts
export const offset: number = 80;

export type CuttingPlanProductionDTO = {
    name: string
    x: number
    width: number
    productionType: string
    info: Map<string, string>
    color: string | null
}

export type CuttingPlanDTO = {
    group: string[]
    productions: CuttingPlanProductionDTO[]
    width: number
    standardWidth: number
    amount: number
    info: Map<string, string>
}

export type SchemaDTO = {
    kpi: Map<string, string>
    cuttingPlans: CuttingPlanDTO[]
}

File: framework/csp1d-interface-renderer/src-tauri/src/dto.rs
use std::collections::BTreeMap;

use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
#[serde(rename_all = "camelCase")]
pub struct CuttingPlanProductionDTO {
    pub name: String,
    pub x: f64,
    pub width: f64,
    pub production_type: String,
    #[serde(default)]
    pub info: BTreeMap<String, String>,
}

#[derive(Serialize, Deserialize, Debug)]
#[serde(rename_all = "camelCase")]
pub struct CuttingPlanDTO {
    pub group: Vec<String>,
    pub productions: Vec<CuttingPlanProductionDTO>,
    pub width: f64,
    pub standard_width: f64,
    pub amount: u64,
    #[serde(default)]
    pub info: BTreeMap<String, String>,
}

#[derive(Serialize, Deserialize, Debug)]
#[serde(rename_all = "camelCase")]
pub struct SchemaDTO {
    #[serde(default)]
    pub kpi: BTreeMap<String, String>,
    pub cutting_plans: Vec<CuttingPlanDTO>,
}
